File: src/main/java/com/EmosewaPixel/pixellib/materialSystem/element/Isotope.java
package com.EmosewaPixel.pixellib.materialsystem.element;

//An isotope is an element with a different amount of neutrons, both in irl and here
public class Isotope extends Element {
    private Element standardIsotope;

    public Isotope(String symbol, Element element, int nucleons) {
        super(symbol, element.getProtons(), nucleons - element.getProtons());
        standardIsotope = element;
    }

    public Isotope(Element element, int nucleons) {
        this(element.getSymbol() + "-" + Integer.toString(nucleons), element, nucleons);
    }

    public Element getStandardIsotope() {
        return standardIsotope;
    }
}

File: src/main/kotlin/com/scientianova/versatile/materialsystem/elements/Isotope.kt
package com.scientianova.versatile.materialsystem.elements

//An isotope is an element with a different amount of neutrons, both in irl and here
class Isotope(symbol: String, val standardIsotope: Element, nucleons: Int) : Element(symbol, standardIsotope.protons, nucleons - standardIsotope.protons) {
    constructor(element: Element, nucleons: Int) : this(element.symbol + "-" + nucleons.toString(), element, nucleons)
}